shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -194,38 +195,34 @@ private JobAttempt newJob(AlluxioURI filePath, int replication) {
   }
 
   /**
-   * Waits one job to complete.
+   * Waits for at least one job to complete.
    */
-  private void waitJob() throws IOException {
-    boolean removed = false;
+  private void waitJob() {
+    AtomicBoolean removed = new AtomicBoolean(false);
     while (true) {
-      Iterator<JobAttempt> iterator = mSubmittedJobAttempts.iterator();
-
-      while (iterator.hasNext()) {
-        JobAttempt jobAttempt = iterator.next();
+      mSubmittedJobAttempts = mSubmittedJobAttempts.parallelStream().filter((jobAttempt) -> {
         Status check = jobAttempt.check();
-
         switch (check) {
           case CREATED:
           case RUNNING:
-            continue;
+            return true;
           case CANCELED:
           case COMPLETED:
-            removed = true;
-            jobAttempt.close();
-            iterator.remove();
-            continue;
-          case FAILED:
-            if (!jobAttempt.run()) {
-              removed = true;
-              iterator.remove();
+            removed.set(true);
+            try {
+              jobAttempt.close();
+            } catch (IOException e) {
+              LOG.warn("Unable to close job master client", e);
             }
-            continue;
+            return false;
+          case FAILED:
+            removed.set(true);
+            return false;
           default:
             throw new IllegalStateException(String.format("Unexpected Status: %s", check));
         }
-      }
-      if (removed) {
+      }).collect(Collectors.toList());
```

Comment:
Was the issue that checking everything in serial was too slow? Because this still waits for all the jobs to be checked.

Reply:
that is correct. Performing listStatus on all 1000 jobs is slow.